hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveMetastoreClientWrapper.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.hive;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.Table;
+
+import org.apache.hudi.common.util.StringUtils;
+import org.apache.hudi.hive.client.HiveShim;
+
+import org.apache.thrift.TException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+import static org.apache.hudi.common.util.ValidationUtils.checkArgument;
+
+/**
+ * Wrapper class for Hive Metastore Client, which embeds a HiveShim layer to handle different Hive versions.
+ * Methods provided mostly conforms to IMetaStoreClient interfaces except those that require shims.
+ *
+ * This code is mainly copy from Apache/Flink.
+ */
+public class HiveMetastoreClientWrapper implements AutoCloseable {
```

Comment:
Few other names I can think of. not too strong on any of them. wanted to hear your thoughts. 
HoodieHiveMetastoreClient
HiveMetastoreClientDelegator.